Doc/library/unittest.mock.rst

```diff
@@ -1307,8 +1307,10 @@ patch
     is patched with a *new* object. When the function/with statement exits
     the patch is undone.
 
-    If *new* is omitted, then the target is replaced with a
-    :class:`MagicMock`. If :func:`patch` is used as a decorator and *new* is
+    If *new* is omitted, then the target is replaced with an
+    :class:`AsyncMock` if the patched object is an async function or
```

Comment:
Is it just async functions? What about async methods? Is there such a thing as an async `__init__` or `__new__`?

Reply:
This is what it is controlling it at the moment: https://github.com/python/cpython/blob/master/Lib/unittest/mock.py#L49 It is checking for `__code__` so it seems to be only for functions at the moment.

[I was wondering about that](https://github.com/python/cpython/pull/13574#discussion_r287576935) for callable class. My main concern was what if I change a function by a callable attribute but it seems there is no good way to check that at the moment. [`asyncio.iscoroutinefunction` works only for functions](https://github.com/python/cpython/pull/13574#discussion_r287586606).

Reply:
Hmm, doesn't seem sensible to make changes to the docs until the rest of this is bottomed out.

Reply:
Note this is landed already and docs are updated in some other places. The documentation for "patch" does not do what it says it does for async functions right now. Also, I am not sure at all this is going to change.

Reply:
This looks like correct wording to me, perhaps we will find a way to make the callable check a little more granular but there is no reason to block this PR on what might be a future change in implementation.